### DISCUSSION TITLE
Translate Organic Maps UI on GitHub

### DISCUSSION THREAD

**Discussion by rtsisyk** [organicmaps/organicmaps#1489] [Translations]
Good morning/afternoon/evening, our beloved users and contributors all around the world! 🗺️

We are desperately looking for your help with translating Organic Maps into other languages. Organic Maps already supports 34 different languages and dialects, and this number is growing:

- ar Arabic
- be Belarusian
- bg Bulgarian
- cs Czech
- da Danish
- de German
- el Greek
- en English (US)
- en_UK English (UK)
- es Spanish
- fa Persian
- fi Finnish
- fr French
- he Hebrew
- hu Hungarian
- id Indonesian
- it Italian
- ja Japanese
- ko Korean
- no a.k.a nb Norwegian
- nl Dutch
- pl Polish
- pt Portuguese
- ro Romanian
- ru Russian
- sk Slovak
- sv Swedish
- sw Swahili
- th Thai
- tr Turkish
- uk Ukrainian
- vi Vietnamese
- zh-cn a.k.a zh-Hans Chinese (simplified)
- zh-tw a.k.a zh-Hant Chinese (traditional)

Please contribute translations for your language to make application better 🍃:

1. Drop a message in comments to this post that you are fluent in `$language` to get your maintainer 🎩.
2. Review [data/strings/strings.txt](https://github.com/organicmaps/organicmaps/blob/master/data/strings/strings.txt) file, correct all inaccuracies and add new translations. No special tools needed - just edit the file on [GitHub](https://github.com/organicmaps/organicmaps/edit/master/data/strings/strings.txt) or by using [Visual Studio Code Online](https://vscode.dev/).
3. Send [Pull Requests](https://github.com/organicmaps/organicmaps/blob/master/docs/CONTRIBUTING.md#translations) and get reviews from other people fluent in `$language`.

Please check more [thorough docs](https://github.com/organicmaps/organicmaps/blob/master/docs/TRANSLATIONS.md) on other translations files, helper scripts, etc.

**Comment by starsep** [comment]
I am fluent in pl Polish, I will review strings.txt file.

**Comment by vbertola** [comment]
I just started to do Italian.

**Comment by vbertola** [comment]
I did the first 6510 lines for Italian and saved the result - I want to continue working in the next few days and produce a single pull request at the end, though I'm not familiar at all with Github (especially via web) so I'm not sure how to do it. In the meantime, I have some questions and comments:

[download_resources] -> is this mandatory or optional? The English / French text seems to use a courtesy form ("please allow us...") while the current Italian translation is much more direct ("it is mandatory to...").

[fuel] -> in some languages this just says "fuel" while in others it says "fuel station", which is the correct one? Same for others in this group, e.g. [tourism]; also, should these terms be singular or plural?

[children] -> this seems to be a weird category for "family holiday", which is not a common concept in my language. I translated it "for children".

[pref_zoom_summary] -> English is "display on the map" but almost all other languages say "display on the screen".

[downloader_downloaded_subtitle], [downloader_available_maps]... -> is this singular or plural? male or female? English has no gender and also no plural in adjectives, but other languages do, so it's vital to know the context.

Colors: I must say this app has a weird choice of colours - e.g. the two grays which are almost impossible to tell apart, and in Italian there are not even common names to express them - same naming problems for "teal" or "lime". I did my best :)

**Reply by biodranik** [reply]
download_resources is mandatory, it downloads the World map.

[fuel] text is insertable when a user selects it from the Search window. It should match synonyms in data/categories.txt. Its meaning is "gas stations" — e.g. find places where you can refuel your car (plural). Or just "Gas" or "Petrol".

The same is for [tourism]: it should find different places interesting for tourists (plural).

pref_zoom_summary is a setting in preferences that toggles +/- buttons on the screen, on top of the map. So the screen sounds better.

Colors are weird indeed :)

**Comment by te2dy** [comment]
I can contribute to the French translation.

**Comment by Supaplextw** [comment]
I can contribute to the zh-tw translation

**Comment by gkalausis** [comment]
I can contribute to Lithuania translation, but I don't see it in the list 😕

**Reply by biodranik** [reply]
You are welcome to add it! Just insert `lt = YourTranslation` everywhere.

**Reply by gkalausis** [reply]
Should I fully translated everything and only then create PR or can it be smaller PR's? Because it takes time to translate everything :)

**Reply by rtsisyk** [reply]
English will be used as a fallback.

**Comment by Kazing** [comment]
I am fluent in French and can contribute to translations. I will review the string.txt file.

**Comment by biodranik** [comment]
There are also few more files to translate:

1. TTS sounds [data/strings/sound.txt](https://github.com/organicmaps/organicmaps/blob/master/data/strings/sound.txt)
2. Map object types [data/strings/types_strings.txt](https://github.com/organicmaps/organicmaps/blob/master/data/strings/types_strings.txt)
3. "What types to search" and search synonyms [data/categories.txt](https://github.com/organicmaps/organicmaps/blob/master/data/categories.txt)
4. Cuisines [data/categories_cuisines.txt](https://github.com/organicmaps/organicmaps/blob/master/data/categories_cuisines.txt)
5. Few iOS specific strings [iphone/plist.txt](https://github.com/organicmaps/organicmaps/blob/master/iphone/plist.txt)
6. Android stores description [android/src/google/play/listings/en-US/](https://github.com/organicmaps/organicmaps/blob/master/android/src/google/play/listings/en-US/)
7. AppStore description [iphone/metadata/en-US/](https://github.com/organicmaps/organicmaps/blob/master/iphone/metadata/en-US/)

**Reply by rtsisyk** [reply]
Please add to CONTRIBUTING.md!!!

**Reply by pastk** [reply]
https://github.com/organicmaps/organicmaps/blob/master/docs/TRANSLATIONS.md

**Comment by LeoAl1590** [comment]
I can help with Spanish

**Comment by matheusgomesms** [comment]
I can glad review/translate strings in Portuguese-Brazil :D

I would also suggest to make available for translation the app screenshots templates (the ones that appear on App/Play Store), if possible. Not sure how it was made, but I believe this is a nice touch to attract non-English speakers ;)

**Comment by jimcarst** [comment]
I can help with the _nl_-translation. Perhaps it's useful for this issue to use a tool for knowing and keeping track of which translations originate from Google Translate and which are manually corrected.

**Comment by houtari** [comment]
In the UI I've bumped into some Finnish phrases that in my ears sound a bit odd, which I'll be glad to fix. I can take a look at the Finnish and Swedish translations as I speak both fluently. Btw. is there a way to change the used language in the UI? If so I haven't found it yet. This would be handy when you do translations - specially in several languages. Is there's a straightforward trick to find possible missing translations within a single language?

**Reply by biodranik** [reply]
That would be great!

1. To switch UI language you need either to switch default system language for iOS and Android or select the app's language in the app's settings on iOS.
2. There is tools/python/find_untranslated_strings.py, but it's outdated and does not do its job very well. It would be great if someone helps fixing it (at least, to properly support plural forms like "en:one", "en:other" etc.

**Reply by pastk** [reply]
@houtari the script to find missing translations had been fixed, e.g.
`tools/python/strings_utils.py --languages fi,sv --missing-translations`
Please check `types_strings.txt` file also if you have time: `tools/python/strings_utils.py --types-strings --languages fi,sv --missing-translations`

**Comment by LaoshuBaby** [comment]
I can contribute to zh-CN/zh-Hans strings.

I'm a native speaker of Simplified Chinese and now living in China. I've already have experience in iD editor's localization in zh-CN, and also contributed strings to taginfo, JOSM, and somethings other.

**Reply by rtsisyk** [reply]
=> https://github.com/orgs/organicmaps/teams/translations-zh-hans

**Comment by matheusgomesms** [comment]
Is this going to be handled in Weblate/any translation tool in the near future or shall we work on strings.txt file directly?

**Comment by vbertola** [comment]
Indeed Github is problematic for those of us who do not use it normally (not everyone here writes code). A couple of weeks ago I spent quite some time to edit strings via the web interface, I saved the result but I did not send a pull request because I wanted to continue working on it and just send one. As a result, apparently my edits (even in my fork of the project) have disappeared completely...

**Reply by biodranik** [reply]
Please contribute in parts, that's ok. We're still experimenting with weblate, unfortunately it doesn't properly handle our current workflow.

**Reply by pastk** [reply]
> Indeed Github is problematic for those of us who do not use it normally (not everyone here writes code). A couple of weeks ago I spent quite some time to edit strings via the web interface, I saved the result but I did not send a pull request because I wanted to continue working on it and just send one. As a result, apparently my edits (even in my fork of the project) have disappeared completely...

Hey, I think I found your lost translations! Is that it?
https://github.com/vbertola/organicmaps/commit/5321dd54093598df685925c867b995d861e296ba

**Reply by vbertola** [reply]
Yes, thanks! But how do I continue editing that file so that I can then submit the complete translations? (also in that version there is an error that I need to fix)

**Reply by pastk** [reply]
> Yes, thanks! But how do I continue editing that file so that I can then submit the complete translations? (also in that version there is an error that I need to fix)

Go here https://github.com/vbertola/organicmaps/blob/patch-1/data/strings/strings.txt
and click the edit button - continue editign, fixing previous things etc.
When saving choose the default option to save to the same branch (your branch is called `patch-1`).
Then you can edit one more time and again and again - its a good idea to save from time to time so that you don't lose your work.
When you think you're finished create a Pull Request from your `patch-1` branch - it will include all your commits / save points. See https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request

**Reply by pastk** [reply]
And actually making a few smaller pull requests instead of one big PR is better. Because someone else with a knowledge of Italian will need to review your changes and this is also easier to do in smaller batches :)

**Comment by houtari** [comment]
Looking at the search category strings to be translated I started wondering why only the first category is phrased as a question "Where to eat". Why isn't the next one also "Where to sleep"? Instead it's named "Hotel", eventhought it lists many other places than just hotels. How about ... "places to eat", "places to sleep" ... etc

**Reply by biodranik** [reply]
This is a very good idea!

**Reply by pastk** [reply]
+1
though I'm not sure it'll be possible to word all the categories in this manner, so we'll have two styles in the same list anyway..
E.g. how do we change
- Police
- Toilet (where to pee and poo? 8D)
- Bank
etc.

E.g. "Where to sleep" is definitely a better alternative for the "Hotel", because it searches not only for hotels, but even for camping grounds!

**Comment by csidirop** [comment]
I am fluent in `de German` and `el Greek` if needed.

**Reply by biodranik** [reply]
Greek translations are missed in many places, and also should be validated.

**Reply by csidirop** [reply]
Whats the supposed way to submit any change? In the opening post @rtsisyk wrote:
> 2. Review data/strings/strings.txt file, correct all inaccuracies and add new translations. No special tools needed - **just edit the file on GitHub** or by using Visual Studio Code Online.
> 3. Send Pull Requests and get reviews from other people fluent in $language.

But following the link in 3. we should 
> run ./tools/unix/generate_localizations.sh and create a Pull Request.

Whether editing the textfiles or building the locs is the same for me. I just can't cant build them on the way (eg. on my phone). And editing the textfiles creates a fork, which I need to submit seperatly:

> You’re making changes in a project you don’t have write access to. We’ve created a fork of this project for you to commit your proposed changes to. Submitting a change will write it to a new branch in your fork, so you can send a pull request. 

Is that how you wanted it?

**Reply by biodranik** [reply]
Please edit source strings files (and don't forget to sign your commits with DCO). If you can, run the generation script. If you can't, NP, we'll run it ourselves.

Regarding forks, yes, Github doesn't allow anyone to write directly into OM repository. So you need to fork the repo first.

**Reply by csidirop** [reply]
Okay, thanks

**Comment by pastk** [comment]
Please check updated [translation docs](https://github.com/organicmaps/organicmaps/blob/master/docs/TRANSLATIONS.md)!

**Comment by j13m126** [comment]
What if I want to add a new language? Can I just add another language, or is there anything else to consider?

**Reply by pastk** [reply]
Yeap there are more changes to do.

- Is it supported by both Android and iOS?
- Does this language need a special font?
- Do you want to add it as a UI language, or a map language or both?

Check PRs for a recent addition of Marathi language (a special font, UI and map use):
https://github.com/organicmaps/organicmaps/pull/2247
https://github.com/organicmaps/organicmaps/pull/2587/files#diff-3d58789860b74a030a907066f95e9b5da8e211ecef340e59035ca4418ca8f99b
https://github.com/organicmaps/organicmaps/pull/2845

**Comment by Markus40** [comment]
I can contribute to the German translation.

**Comment by palzkillm** [comment]
I can also help with german.

**Comment by rtsisyk** [comment]
Hello! Thanks for contributing localizations. I created translation-$lang teams on GitHub for each language and invited all of yours. We are also testing CODEOWNERS setting to get you notified on any change in files with your language. 

https://github.com/organicmaps/organicmaps/pull/3161

**Reply by jimcarst** [reply]
I'm getting a lot of notifications with review requests. For each it takes some time to figure out for what review is required. Perhaps it'd be good to limit these review requests to only modifications of strings.txt commits, so you don't need to read the whole pull request discussion?

**Comment by d4f5409d** [comment]
Please consider using different folders for different languages like Signal does, because this is a mess right now to edit. Or allow user to translate through Weblate. Thank you.

**Reply by biodranik** [reply]
Implementing proper integration with Weblate requires a lot of our time. An improper integration with Weblate will take a lot of our time to support.

As our resources are very limited, we try to focus on the most important bugs and features ATM. Hopefully, we'll make it more convenient someday.

**Reply by d4f5409d** [reply]
Take your time. I'm glad I became your supporters.

**Comment by DaN0mic** [comment]
I'm fluent in de German

**Reply by rtsisyk** [reply]
Invited you to @organicmaps/translations-de

**Comment by rtsisyk** [comment]
@organicmaps/translations-de, we want to start providing Deutsche translations for Google Play / F-Droid / AppStore listing and changelogs starting from the next release (in mid of Sept). Your help is needed with translation of title, description and other texts:

- [Android](https://github.com/organicmaps/organicmaps/tree/master/android/src/fdroid/play/listings)
- [iOS](https://github.com/organicmaps/organicmaps/tree/master/iphone/metadata)

Could you please take a look? Please don't worry about git and github. We will help you with it. In worst case you can post texts just here.

Also, what communication tool do you prefer to coordinate work?

- GitHub Discussions (here)
- Matrix
- Telegram
- Email
- Anything else?

**Reply by csidirop** [reply]
I have briefly read over the texts and can translate them today after work (or maybe at my break). We're only talking about the linked texts, right?

Github is fine for me, but I'd appreciate it if someone looked at it first.

**Reply by rtsisyk** [reply]
>  We're only talking about the linked texts, right?

Yes.

**Android**:

- [full-description.txt](https://github.com/organicmaps/organicmaps/blob/master/android/src/fdroid/play/listings/en-US/full-description.txt)
- [short-description.txt](https://github.com/organicmaps/organicmaps/blob/master/android/src/fdroid/play/listings/en-US/short-description.txt)
- [title.txt](https://github.com/organicmaps/organicmaps/blob/master/android/src/fdroid/play/listings/en-US/title.txt)

**iOS**:

- [name.txt](https://github.com/organicmaps/organicmaps/blob/master/iphone/metadata/en-US/name.txt)
- [subtitle.txt](https://github.com/organicmaps/organicmaps/blob/master/iphone/metadata/en-US/subtitle.txt)
- [description.txt](https://github.com/organicmaps/organicmaps/blob/master/iphone/metadata/en-US/description.txt)
- [promotional_text.txt](https://github.com/organicmaps/organicmaps/blob/master/iphone/metadata/en-US/promotional_text.txt)

Please note that those fields are limited by length. We have the automatic check on GitHub that verifies it. The language code is `de-DE` both for Apple and Google.

>  I'd appreciate it if someone looked at it first.

Just create a PR and other people will be able to add comments and suggestions. 
🙏

**Reply by csidirop** [reply]
> Please note that those fields are limited by length. We have the automatic check on GitHub that verifies it. The language code is `de-DE` both for Apple and Google.

Where is the limit?

Edit: 
- fdroid/play/listings/.../short-description.txt: **limit at 80**
- fdroid/play/listings/.../title.txt: **limit at 50**
- iphone/.../name.txt: **limit at 30**
- iphone/.../promotional_text.txt: **limit at 170**
- iphone/.../keywords.txt: **limit at 100**

**Reply by csidirop** [reply]
And don't we need an additional translation here: [android/src/google/play/listings](https://github.com/organicmaps/organicmaps/tree/master/android/src/google/play/listings)?

**Reply by csidirop** [reply]
I finished the android (fdroid only?) translations. Here my draft, if anyone wants to look over it: https://github.com/organicmaps/organicmaps/pull/3351

IOS store is coming later.

**Reply by d4f5409d** [reply]
I am hu-HU translator. Why am I in here?

**Reply by rtsisyk** [reply]
@d4f5409d , you were not in Deutsche translation group. I invited you to https://github.com/orgs/organicmaps/teams/translations-hu.

**Reply by rtsisyk** [reply]
> And don't we need an additional translation here: [android/src/google/play/listings](https://github.com/organicmaps/organicmaps/tree/master/android/src/google/play/listings)?

we usually just copy them

**Reply by rtsisyk** [reply]
https://play.google.com/store/apps/details?id=app.organicmaps&hl=de German translation is in the store! Thanks all!

**Comment by rtsisyk** [comment]
We implemented initial support for multiple languages for the website:

- https://organicmaps.app/tr/ - Türkçe
- https://organicmaps.app/it/ - Italiano 

To translate to your language, just copy `_index.md` into `_index.<LANGUAGE>.md` and add your translation:

https://github.com/organicmaps/organicmaps.github.io/tree/master/content

**Comment by Misalf-git** [comment]
Hello,

I currently have no access to a PC and I cannot edit strings.txt on mobile.

Maybe someone would look at the german translation for "News" on the Help screen (question mark button).
It reads "Nachrichten", which is practically correct for the term allone, but not so much for the context.
I'd suggest to change it to "News" as this anglicism	 is often used in german. Another, very correct option would be "Neuigkeiten" but I'd go for "News".

**Reply by palzkillm** [reply]
Aktuelles?
Neuigkeiten?
Neues?

**Reply by palzkillm** [reply]
But I concur - News would be acceptable in german as well.

**Reply by palzkillm** [reply]
Simply Neu? While the term itself translates to just New, but it‘s often used in any context where news are listed.

**Reply by biodranik** [reply]
Which translation is used in the similar context on your phones in other apps?

**Reply by Misalf-git** [reply]
![Screenshot_20220917-173618_crDroid Home.png](https://user-images.githubusercontent.com/72605561/190864809-a4354962-ed9c-442d-a78b-747f2b409666.png)

**Reply by csidirop** [reply]
I think _news_ is quite common, but @palzkillm suggestions (_Aktuelles_ and _Neuigkeiten_) are also good.

Firefox:
![grafik](https://user-images.githubusercontent.com/43964592/190964531-961bd4d1-bc7c-41a2-83bd-16ff3b24e255.png)
Steam:
![grafik](https://user-images.githubusercontent.com/43964592/190964230-dbb31503-6cb1-4766-b8f4-9a6c1c3268fe.png)

**Reply by Misalf-git** [reply]
Like above screenshot shows ("Feedback", "Update", "Community"), certain terms, especially regarding software, simply don't need translation into german nowadays.

**Reply by biodranik** [reply]
Thanks! Let's use News: #3483

**Comment by Gnorv** [comment]
I can help with German - DE translations. I found some comments about a translation group but clicking the link sent me to a non [existing github page](https://github.com/orgs/organicmaps/teams/translations-de).

**Reply by biodranik** [reply]
@rtsisyk can you please help?

**Reply by rtsisyk** [reply]
> I can help with German - DE translations. I found some comments about a translation group but clicking the link sent me to a non [existing github page](https://github.com/orgs/organicmaps/teams/translations-de).

Invited you to the team. Please check email.

**Comment by s8321414** [comment]
I can translate to zh-tw, please add me to translations-zh-tw team, thank you :smiley:

**Reply by rtsisyk** [reply]
Hi @ s8321414, I created a separate @organicmaps/translations-zh-hant team and invited you. I hope it works.

**Reply by s8321414** [reply]
> Hi @ s8321414, I created a separate @organicmaps/translations-zh-hant team and invited you. I hope it works.

Thank you!

**Comment by Supaplextw** [comment]
I can translate to zh-hant, please add me to translations-zh-hant team, thank you :)

**Reply by rtsisyk** [reply]
Invited.

**Comment by ghost** [comment]
Hi Roman (@rtsisyk),

Completed strings.txt and categories.txt translation to Estonian (et) locale.

Also sent a pull request.

Kind regards,
Aulo

**Reply by rtsisyk** [reply]
> Hi Roman (@rtsisyk),
> 
> Completed strings.txt and categories.txt translation to Estonian (et) locale.
> 
> Also sent a pull request.
> 
> Kind regards, Aulo

Added you to the group of Estonian translators! Thanks!

**Comment by Mr-Mime** [comment]
If you want more translators for `german` (native) and/or `spanish` (fluently) just add me to the needed group. I am happy to help.

**Reply by rtsisyk** [reply]
> If you want more translators for `german` (native) and/or `spanish` (fluently) just add me to the needed group. I am happy to help.

Added you to the group.

**Comment by dvrsaran** [comment]
I can translate it into **Tamil** language.

**Reply by rtsisyk** [reply]
> I can translate it into **Tamil** language.

Please go head ;) Language code is `ta`. https://github.com/organicmaps/organicmaps/blob/master/data/strings/strings.txt

**Comment by rtsisyk** [comment]
@organicmaps/translations-de, I started translating web-site to German. I know that you are  thedream team and fix my broken automatic translation very quickly ;)

https://github.com/organicmaps/organicmaps.github.io/pull/69

**Comment by rtsisyk** [comment]
@organicmaps/translations-fr, please help us with translating web site:

https://github.com/organicmaps/organicmaps.github.io/pull/74

**Comment by rtsisyk** [comment]
Let's talk about moving website into Weblate to make translations easier: https://github.com/organicmaps/organicmaps/discussions/4076#discussion-4655579

**Comment by rtsisyk** [comment]
It would be nice if somebody can volunteer to translate Organic Maps website and app to Hindi. I realized that English is not such widespread in India as I expected. There are too many regional languages such as Tamil, Marathi, Bengali, Urdu, Kannada, and others. Majority of people usually know at least two or even three languages. My expectations that Hindi + English can cover 60-70% of people. Please correct me if I am wrong.

**Reply by proxygeek** [reply]
I'd love to translate it to Hindi and can get started this weekend. OrganicMaps is one of my favorite ones and would love to help out.
(Btw, I know it's planned - but being able to record / export / import a track - is one of the most frequently used features by me, which is what is keeping me from using OM more often).

**Comment by JohannesDeml** [comment]
If there are any more people needed for German, I'm happy to help :)

**Reply by rtsisyk** [reply]
I've invited you. It looks like that we already need a coordinator for German team, 7 people already.

**Comment by Karalynx** [comment]
I am fluent in Lithuanian and have 2 years of experience in translating.

**Reply by rtsisyk** [reply]
I've created @organicmaps/translations-lt team and invited you.

**Comment by tkbremnes** [comment]
I can help out translating into Norwegian $nb
I've got a PR in the making filling some holes, as well as some adjustments to existing strings.

**Reply by rtsisyk** [reply]
I've invited you into @organicmaps/translations-nb team. You will be first!

**Comment by rtsisyk** [comment]
Let's continue about Weblate here: https://github.com/organicmaps/organicmaps/discussions/4515. Please take a look!

**Reply by proxygeek** [reply]
While the discussions continue on Weblate and a decision is pending, are there some translations I can still offer directly in the strings.txt file - especially for Hindi?

**Reply by j13m126** [reply]
Hi @proxygeek , I opened a [pull request](https://github.com/organicmaps/organicmaps/pull/4266) for Hindi a few weeks ago. It would be great if you could have a look at the open suggestions and review it.

**Reply by rtsisyk** [reply]
yes, please follow existing process for now.

**Comment by NLBRT** [comment]
I'm native in Bengali `[bn]` and can contribute to translations. I'll fix up any translation errors from previous editors/translation attempts and continue with the unfinished portions.

**Reply by biodranik** [reply]
Let's start with adding translations to strings.txt

**Comment by j9d3it** [comment]
Hi I am fluent in Japanese, please add me to the maintainers. I've noticed a lot of mistakes.

**Reply by biodranik** [reply]
@j13m126 I can add you to our translation reviewers group for Japanese. You can create a PR with fixes at any time, or create an issue that lists what should be fixed.

**Reply by j13m126** [reply]
I think you meant @j9d3it

**Reply by j9d3it** [reply]
Thank you. I think using something like [Crowdin](https://crowdin.com/page/open-source-project-setup-request) may be good in the future. It's a nice platform for translating software and websites and is free for open source software. They have a Global TM (translation memory) which allows fast translations of common words so that the more difficult sentences can be focused on. It also presents one word or phrase at a time which is easier than editing a big file and messing something up.

**Reply by biodranik** [reply]
We have already discussed it in the past. Crowdin is better than Weblate, as it supports editing markdown files that are used on our website.

The disadvantage is that there are not so many users there, so we need to tell translators to register there.

It would be great if someone can try to setup a demo/trial project with [our website markdown files](https://github.com/organicmaps/organicmaps.github.io/) and check if it's possible to integrate OM's workflow with strings translations and their reviewing.

**Reply by j9d3it** [reply]
You can log in to Crowdin with Github, no need to make a new account. https://accounts.crowdin.com/login

**Reply by biodranik** [reply]
What we need from Crowdin or another system for our website translations:
1. The source en language for content automatically comes from our Github.
2. Any translations are contributed back as markdown files.
3. If already translated en content was changed, or a new en paragraph was added, translators should be notified/aware of it and update the translation.

**Reply by j9d3it** [reply]
1. [They support either file or string based translation with the Github integration.](https://support.crowdin.com/search/?q=github)
2. [Markdown is supported](https://support.crowdin.com/supported-formats/)
3. [Translators can sign up to be notified when new strings are added. I think a changed string counts as a new one.](https://support.crowdin.com/account-settings/#notifications)

**Comment by fitojb** [comment]
I'd like to start gradually translating the app to Asturian. It's my fifth language and I'm becoming fluent in it:)

**Reply by biodranik** [reply]
Not sure that our current localization system will support pdt language code. Do iOS and Android support it natively? With which locale/code?

**Comment by CharliePlett** [comment]
Hi I'm fluent in Plautdietsch (pdt) and I would like to contribute translations to the app.

**Reply by biodranik** [reply]
Not sure that our current localization system will support pdt language code. Do iOS and Android support it natively? With which locale/code?

**Comment by schembriaiden** [comment]
Hiya, I am fluent in Maltese, are you considering adding it to your list?

**Reply by biodranik** [reply]
You can check existing files in data/strings/ and add mt translation there.

**Comment by muralito** [comment]
Galego should be added, to improve the support for this languages and try to reduce the level of conflict with toponymy in Spain. Euskara an Catala are already supported.

**Comment by danamoh0mmed** [comment]
Hi, I'm fluent in Central Kurdish (CKB) "کوردیی ناوەندی" and really excited to translate this app for CKB speakers. Please guide me.

**Reply by biodranik** [reply]
Hi and welcome! Here are the instructions: https://github.com/organicmaps/organicmaps/blob/master/docs/TRANSLATIONS.md

**Comment by carlmorris** [comment]
Please can you add me as a maintainer for Cymraeg (cy)?

Thanks for your consideration, and for your hard work on a great app.

**Reply by biodranik** [reply]
Invitation sent.
